broker/data-access-layer/iaas/src/AwsClient.js

```diff
@@ -0,0 +1,308 @@
+        Key: tagKey,
+        Value: options.tags[tagKey]
+      }))
+    });
+    return Promise.try(() => {
+      return this.blockstorage
+        .createVolume({
+          AvailabilityZone: _.isArray(zones) ? zones[0] : zones,
+          SnapshotId: snapshotId,
+          VolumeType: _.get(options, 'type', 'gp2'),
+          TagSpecifications: tags
+        })
+        .promise();
+    })
+      .then(volume => {
+        const describeReq = {
+          VolumeIds: [volume.VolumeId]
+        };
+        return this.blockstorage.waitFor('volumeAvailable', describeReq).promise();
+      })
+      .then(volResponse => volResponse.Volumes[0])
+      .then(volume => {
+        const responseTags = volume.Tags || [];
+        const outTags = {};
+        _.forEach(responseTags, tag => {
+          outTags[tag.Key] = tag.Value;
+        });
+        return {
+          volumeId: volume.VolumeId,
+          size: volume.Size,
+          zone: volume.AvailabilityZone,
+          type: volume.VolumeType,
+          extra: {
+            type: volume.VolumeType,
+            tags: outTags
+          }
+        };
+      });
+  }
+
+  getDiskMetadata(diskId) {
+    return Promise.try(() => {
+      return this.blockstorage
+        .describeVolumes({
+          VolumeIds: [diskId]
+        })
+        .promise();
+    })
+      .then(diskResponse => diskResponse.Volumes[0])
+      .then(volume => {
+        const responseTags = volume.Tags || [];
+        const outTags = {};
+        _.forEach(responseTags, tag => {
+          outTags[tag.Key] = tag.Value;
+        });
+        return {
+          volumeId: volume.VolumeId,
+          size: volume.Size,
+          zone: volume.AvailabilityZone,
+          type: volume.VolumeType,
+          extra: {
+            type: volume.VolumeType,
+            tags: outTags
+          }
+        };
+      });
+  }
+
+  getContainer(container) {
+    if (arguments.length < 1) {
+      container = this.containerName;
+    }
+
+    let s3Options = {
+      Bucket: container
+    };
+
+    return this.storage.
+      listObjectsAsync(s3Options)
+      .then(data => {
+        return data;
+      });
+  }
+
+  list(container, options) {
+    if (arguments.length < 2) {
+      options = container;
+      container = this.containerName;
+    }
+    let s3Options = {
+      Bucket: container
+    };
+
+    if (options.marker) {
+      s3Options.Marker = options.marker;
+    }
+
+    if (options.prefix) {
+      s3Options.Prefix = options.prefix;
+    }
+
+    if (options.maxKeys) {
+      s3Options.MaxKeys = options.maxKeys;
+    }
+
+    return this.storage
+      .listObjectsAsync(s3Options)
+      .then(data => {
+        let isTruncated = data.IsTruncated;
+        let list = data.Contents;
+
+        list.forEach(element => {
+          element.name = element.Key;
+          element.lastModified = element.LastModified;
+          delete element.Key;
+          delete element.LastModified;
+        });
```

Comment:
Performed these operations to match the output of `CloudProviderClient's list` method.